### DISCUSSION TITLE
Async defaultValues and warning message

### DISCUSSION THREAD

**Discussion by Sossialy** [react-hook-form/react-hook-form#2862] [General] [answered]
Hello community :)

First of all, thanks for the library, it truly rocks!

The situation:
- Imagine a big form on a user account page, with all its information.
- The user information is either in my cache or I get it asynchronously via an API call on component mount.
- So I use a defaultValues with either the user or `{}` plus a reset (when async) for my big form.
- I got warning messages when data is async because defaultValues is set to `{}`. Like `lastname is missing in the 'defaultValue' prop of either its Controller (https://react-hook-form.com/api#Controller) or useForm (https://react-hook-form.com/api#useForm)`

What I would like:
- I know defaultValues is set only when I have the data and I don't want warning messages in my console regarding those asynchronously fetched data.
- I don't want to manually set mock defaults when the data is async, this would slow me down every-time a new field has to be added.

It'll be super nice if you guys found a solution to this problem, maybe I missed something.
Best,
Luc

**Comment by bluebill1049** [comment, marked as answer]
`useEffect(() => reset(), [])` should do it for you.

**Reply by Sossialy** [reply]
Hello Bluebill,

As explained, sometime my data is async and I'm using reset to use them as defaultValues, the problem is that the warning message is displayed before the data arrives because defaultValues are empty at that moment.

**Reply by bluebill1049** [reply]
can you reproduce that in a codesandbox?

**Reply by Sossialy** [reply]
Here you go: https://codesandbox.io/s/beautiful-moore-p9pt6?file=/src/App.js :)

**Reply by Sossialy** [reply]
Plus we have that error message regarding uncontrolled => controlled component, which I got used to but if you know how to get rid of it I'll gladly listen ^^

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/vibrant-river-csw24

or useFrom's defaultValues

**Reply by Sossialy** [reply]
But then if `user` is in cache the controller won't take it as defaultValue right?
![image](https://user-images.githubusercontent.com/59333340/92710720-52338080-f358-11ea-93bf-90251b947da7.png)

**Reply by bluebill1049** [reply]
no, reset will always update everything in the form

**Reply by Sossialy** [reply]
Yes, will put defaultValue on every Controller then!
Thanks you very much :)

**Reply by bluebill1049** [reply]
np :)
